pkg/util/initializer.go

```diff
@@ -0,0 +1,60 @@
+package util
+
+import (
+	"context"
+	"sync"
+)
+
+type Initializer struct {
```

Comment:
Not sure I'm following everything going on here - but I think I get the gist.
A new mechanism to control global state related to controller config/settings.

Reply:
TLDR; its a synchronization primitive wrapper that ensures its closure is only set once, and anyone that shows interest in what it encapsulates must wait for it to be set. It was used a lot in opni